<launch>
    <arg name="task" default="goto" />

    <arg name="rviz_config_path" default="$(find ct_ros_nodes)/config/HyA/dms/rviz/hya_dms.rviz"/>

    <node name="rviz" pkg="rviz" type="rviz" args="--display-config $(arg rviz_config_path)" respawn="false" output="screen"/>  
    <param name="hya_description" textfile="$(find ct_models)/urdf/HyA/HyA.urdf" />
  
    <node pkg="robot_state_publisher" type="robot_state_publisher" name="hya_state_publisher" >
      <remap from="robot_description" to="hya_description" />
    </node>
    
    <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" />
    
    <node name="hya_dms" pkg="ct_ros_nodes" type="HyADMS" respawn="false" output="screen">
        <param name="useMultiThread" value="true" />
        <param name="useCodegenModel" value="true" />
        <param name="skip" value="5" />
        <param name="workingDirectory" value="$(find ct_ros_nodes)/config/HyA/dms/$(arg task)" />
    </node>
  
</launch>